### PR TITLE
RVT: Improved polyline conversions

### PR DESCRIPTION
**Pull request by AlanRynne**
## Description

- Fixes issue in Revit where Polyline's with very short segments would completely fail to convert.
- Fixes an issue were we were assuming a `Curve` in a cast, but should have been an `IDisplayable<Polyline>` instead.
- Adds length calculation to `Line` class constructor. This is used during polyline creation to ensure no short segments are converted to Revit.

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## How has this been tested?

- Manual Tests (please write what did you do?)

Original offending commit: https://speckle.xyz/streams/35cc66339a/commits/cb564be587

## Docs

- No updates needed

### CODE REVIEW

File: Objects/Converters/ConverterRevit/ConverterRevitShared/Partial Classes/ConvertGeometry.cs [hunk 1]
```diff
@@ -468,18 +469,37 @@ public CurveArray PolylineToNative(Polyline polyline)
...
+          {
+            Report.LogConversionError(e);
+          }
         }
 
         if (polyline.closed)
         {
-          var speckleLine = new Line(new double[] { pts[pts.Count - 1].x, pts[pts.Count - 1].y, pts[pts.Count - 1].z, pts[0].x, pts[0].y, pts[0].z }, polyline.units);
-          curveArray.Append(LineToNative(speckleLine));
+          var speckleLine = new Line(pts[pts.Count - 1], pts[0] , polyline.units);
```
Comment (clairekuang) [inline Objects/Converters/ConverterRevit/ConverterRevitShared/Partial Classes/ConvertGeometry.cs:494]: Should handle the case here where the polyline first and last points are the same and `.closed() = true`
The new Line may throw an exception in this case?

Comment (izzylys) [reply Objects/Converters/ConverterRevit/ConverterRevitShared/Partial Classes/ConvertGeometry.cs:494]: fair, tho in theory this shouldn't happen ((the polyline class asks for first and last points to be unique tho this is not enforced so someone could easily do this if they don't read the docstring))

**Comment by AlanRynne** [comment]
Since the same logic was being used in 3 parts of the same method, I refactored it out into the ConversionUtils file.

I also added some XML comments to the new functions, and to the `PolylineToNative` to make it clear that the geometry can be modified when small segments are present.

Screenshot 2022-05-25 at 22 04 09

**Review by izzylys — APPROVED** [review]
nice! I like the refactoring the logic out to the utils and I appreciate your helpful xml comments 😊

**Pull request merged**